lib/cyclopedio/mapping/wikipedia_category_utils.rb

```diff
@@ -0,0 +1,67 @@
+require 'cyclopedio/syntax'
+require 'wiktionary/noun'
+
+module Cyclopedio
+  module Mapping
+    # This class provides methods for processing category names and its parsed trees.
+    class WikipediaCategoryUtils
```

Comment:
I propose decorators for categories and articles. singularize_name should be in Nouns or Nouns decorator.